shared/ui/combobox/combobox.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { ChangeEvent, FC, MouseEventHandler, useState } from 'react';
+import { clsx } from 'clsx';
+
+import { Combobox as ComboboxUI } from '@headlessui/react';
```

Comment:
`Primitive` or `Headless` is more convinient name than `ComboboxUI`(it is often called just that).

Reply:
What to name it then if we use more components from an external library? It seems like `Primitive` or `Headless` doesn't make it clear what is being used

Reply:
First of all, `Primitive` is simply the name of an object that represents the namespace for the primitive components it contains.
It's hard for me to imagine that you might want to mix multiple non-styled library components in one component at once. Since they usually have the necessary primitives out of the box to build a complete solution. More likely you will need to embed an already implemented component into the primitive to inherit its logic and styles, rather than writing everything again (as with the `asChild` prop for Radix). Moreover, since different libraries are written by different commands, they are most often incompatible. And you'll again have to write your own "adapters" to make them compatible. Which means you'll still have to describe your implementations on top of the primitives.

But if you are tempted to put several primitive libraries into one component at once, which may cause a `Primitive` namespace confict, this is a signal that you should first write a basic implementation of the stylized primitives, and then include them in the current component.

So there should be no question of name space overlap.

---
About the *"semantic content"* of the names `Primitive` and `ComboboxUI`. They both do not provide any information about what library is used under the hood (because there is no `ComboboxUI` library). And I will have to look at the imorts in the file one way or another to understand where this solution is taken from. But, for example, for me at a quick glance `ComboboxUI` means that this is some implementation in our code base, not a template from the third-party library.
On the whole, this is not a tragedy. But on our project everyone calls primitives whatever he wants, i.e. we have no agreement on naming templates from third-party libraries. And I stuck to this question to create an agreement for uniformity of naming in the project. And as it seems to me, `Primitive` or `Headless` is not bad for describing any library of non-styled components.